### PR TITLE
[webgpu] Add select and fill for prelu

### PR DESCRIPTION
**Pull request by axinging**
This is the first part to fix https://github.com/tensorflow/tfjs/issues/2360.

Performance data about different workPerThread  and workGroupSize.
workPerThread = 4; workGroupSize: [number, number, number] = [16, 1, 1];(The one used in this commit)
'Mean time: 13.611 ms -> 0.272 / rep'
'Mean time: 13.623 ms -> 0.272 / rep'
'Mean time: 13.967 ms -> 0.279 / rep'

workPerThread = 1; workGroupSize: [number, number, number] = [16, 1, 1];
'Mean time: 13.205 ms -> 0.264 / rep'
'Mean time: 13.981 ms -> 0.280 / rep'
'Mean time: 13.644 ms -> 0.273 / rep'

workPerThread = 1; workGroupSize: [number, number, number] = [4, 4, 4];
'Mean time: 13.520 ms -> 0.270 / rep'
'Mean time: 13.910 ms -> 0.278 / rep'
'Mean time: 13.851 ms -> 0.277 / rep'

workPerThread = 1; workGroupSize: [number, number, number] = [4, 4, 1];
'Mean time: 13.965 ms -> 0.279 / rep'
'Mean time: 13.449 ms -> 0.269 / rep'
'Mean time: 13.576 ms -> 0.272 / rep'

workPerThread = 1; workGroupSize: [number, number, number] = [1, 1, 1];
'Mean time: 13.569 ms -> 0.271 / rep'
'Mean time: 13.818 ms -> 0.276 / rep'
'Mean time: 13.449 ms -> 0.269 / rep'

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/2412)

### CODE REVIEW

**Comment by axinging** [comment]
@annxingyuan , PTAL

File: tfjs-backend-webgpu/src/kernels/fill_webgpu.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import {computeDispatch, flatDispatchLayout} from '../webgpu_util';
+import {WebGPUProgram} from './webgpu_program';
+
+export class FillProgram implements WebGPUProgram {
+  variableNames: string[];
+  outputShape: number[] = [];
+  userCode: string;
+  dispatchLayout: {x: number[]};
+  dispatch: [number, number, number];
+  workPerThread = 4;
```
Comment (annxingyuan) [inline tfjs-backend-webgpu/src/kernels/fill_webgpu.ts]: Just curious, did you test other values for workPerThread and workGroupSize? Are these optimal for performance?

File: tfjs-backend-webgpu/src/kernels/fill_webgpu.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+import {util} from '@tensorflow/tfjs-core';
+import {computeDispatch, flatDispatchLayout} from '../webgpu_util';
+import {WebGPUProgram} from './webgpu_program';
+
+export class FillProgram implements WebGPUProgram {
+  variableNames: string[];
```
Comment (annxingyuan) [inline tfjs-backend-webgpu/src/kernels/fill_webgpu.ts]: initialize this to []

File: tfjs-backend-webgpu/src/shader_preprocessor.ts [hunk 3]
```diff
@@ -79,26 +79,29 @@ export function makeShader(
...
-  program.variableNames.forEach((x, i) => {
-    uniformDeclaration += `${getCoordsDataType(inputInfo[i].shape.length)} ${
-        x.charAt(0).toLowerCase() + x.slice(1)}Shape; `;
-    prefixSnippets.push(`
-      layout(std430, set = 0, binding = ${1 + i}) readonly buffer ssb${x} {
-        ${mapToGlslTypes(inputInfo[i].dtype)} ${x}[];
-      };
-    `);
-  });
+  const variableNamesIsArray = Array.isArray(program.variableNames);
```
Comment (annxingyuan) [inline tfjs-backend-webgpu/src/shader_preprocessor.ts]: I think as long as you initialize variableNames in the FillProgram to [], you shouldn't need this boolean.

File: tfjs-backend-webgpu/src/kernels/fill_webgpu.ts (hunk 1 above)
Comment (axinging) [reply tfjs-backend-webgpu/src/kernels/fill_webgpu.ts]: I add performance data of five different workPerThread and workGroupSize (In the original comment). I didn't see any significant differences on macOS Mojave 10.14.4/Intel Iris Plus Graphics 655.

**Pull request merged**